File: C++/606. Construct String from Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    string tree2str(TreeNode* t) {
        string ans = "";
        if(t==nullptr)
            return ans;
        ans += to_string(t->val);
        if(t->left || t->right)
        {
            ans += "(" + tree2str(t->left) + ")";
            if(t->right)
                ans += "("+ tree2str(t->right) + ")";
        }
        return ans;
        
    }
};

File: Python/606. Construct String from Binary Tree.py
class Solution:
    def tree2str(self, t):
        """
        :type t: TreeNode
        :rtype: str
        """
        if t is None:
            return ""
        ans = str(t.val)
        if t.left or t.right:
            ans += "(" + self.tree2str(t.left) + ")"
            if t.right:
                ans += "(" + self.tree2str(t.right) + ")"
        return ans